File: php/testcode/ldapi_002.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ldapi_search_escaped
function ldapi002(BenchmarkRequest $req): BenchmarkResponse {
    $input = $req->param('username');
    $conn = ldap_connect('ldap://localhost');
    $base = 'dc=example,dc=com';
    $safe = ldap_escape($input, '', LDAP_ESCAPE_FILTER); // vuln-code-snippet safe-line php_ldapi_search_escaped
    $result = ldap_search($conn, $base, "(uid=" . $safe . ")");
    $entries = ldap_get_entries($conn, $result);
    return BenchmarkResponse::json($entries);
}
// vuln-code-snippet end php_ldapi_search_escaped

File: ruby/testcode/ldapi_002.rb
require_relative 'shared'

module Net
  module LDAP
    def self.new(options = {}); options; end
    def self.search(options = {}); []; end
    module Filter
      def self.eq(attr, value); "(#{attr}=#{value})"; end
    end
  end
end

# vuln-code-snippet start ruby_ldapi_filter_eq
def lookup_user_filtered(req)
  username = req.param('username')
  ldap = Net::LDAP.new(host: 'ldap.example.com')
  safe_filter = Net::LDAP::Filter.eq('uid', username)  # vuln-code-snippet safe-line ruby_ldapi_filter_eq
  entries = ldap.search(filter: safe_filter)
  BenchmarkResponse.json({ count: entries.length })
end
# vuln-code-snippet end ruby_ldapi_filter_eq
